src/WebJobs.Script/Scale/HostPerformanceManager.cs

```diff
@@ -27,20 +39,99 @@ public HostPerformanceManager(IEnvironment environment, IOptions<HostHealthMonit
 
             _environment = environment;
             _healthMonitorOptions = healthMonitorOptions;
+            _logger = logger;
+            _serviceProvider = serviceProvider;
+            _processMonitor = new ProcessMonitor(Process.GetCurrentProcess().Id, environment);
+
+            _processMonitor.Start();
         }
 
-        public virtual bool IsUnderHighLoad(Collection<string> exceededCounters = null, ILogger logger = null)
+        /// <summary>
+        /// Check sandbox enforced performance counters
+        /// </summary>
+        public virtual bool PerformanceCountersExceeded(Collection<string> exceededCounters = null, ILogger logger = null)
         {
             var counters = GetPerformanceCounters(logger);
             if (counters != null)
             {
-                return IsUnderHighLoad(counters, exceededCounters, _healthMonitorOptions.Value.CounterThreshold);
+                return PerformanceCounterThresholdsExceeded(counters, exceededCounters, _healthMonitorOptions.Value.CounterThreshold);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Check both sandbox enforced performance counters as well as process level thresholds
+        /// like CPU, memory, etc.
+        /// </summary>
+        public virtual async Task<bool> IsUnderHighLoad(Collection<string> exceededCounters = null, ILogger logger = null)
+        {
+            return PerformanceCountersExceeded(exceededCounters, logger) ||
+                await ProcessThresholdsExceeded(exceededCounters, logger);
+        }
+
+        private async Task<bool> ProcessThresholdsExceeded(Collection<string> exceededCounters = null, ILogger logger = null)
```

Comment:
This new process monitoring is the key for the new FE admin/host/ping scale ping handling. It enables us to return a status to influence scale out, that takes into account the current host and worker cpu load (and perhaps other metrics in the future).